internal/backend/connectionsinitsvc/svc.go

```diff
@@ -0,0 +1,157 @@
+		http.Error(w, "connection not found", http.StatusNotFound)
+		return
+	}
+
+	integ, err := s.svcs.Integrations.GetByID(r.Context(), conn.IntegrationID())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if !integ.IsValid() {
+		http.Error(w, "integration not found", http.StatusNotFound)
+		return
+	}
+
+	http.Redirect(w, r, fmt.Sprintf("%s?cid=%v&origin=%s", integ.ConnectionURL(), cid, origin), http.StatusFound)
+}
+
+// postInit is the last step in the connection initialization flow.
+// It saves the connection's data in the connection's scope, and
+// redirects the user to the last HTTP response, based on its origin
+// (local AK server / local VS Code extension / SaaS web UI).
+func (s Svc) postInit(w http.ResponseWriter, r *http.Request) {
+	vars, origin := r.URL.Query().Get("vars"), r.URL.Query().Get("origin")
+
+	var data []sdktypes.Var
+	if err := kittehs.DecodeURLData(vars, &data); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	cid, err := sdktypes.StrictParseConnectionID(r.PathValue("id"))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	data = kittehs.Transform(data, func(v sdktypes.Var) sdktypes.Var {
+		return v.WithScopeID(sdktypes.NewVarScopeID(cid))
+	})
+
+	if err := s.svcs.Vars.Set(r.Context(), data...); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var u string
+	switch origin {
+	case "vscode":
+		u = "vscode://autokitteh.autokitteh?cid=%s"
+	default:
+		// Another redirect just to get rid of the secrets in the URL.
+		u = "/connections/%s/success"
+	}
+	http.Redirect(w, r, fmt.Sprintf(u, cid), http.StatusFound)
+}
+
+func (s Svc) testConnection(w http.ResponseWriter, r *http.Request) {
+	id := r.PathValue("id")
+
+	cid, err := sdktypes.StrictParseConnectionID(id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	st, err := s.svcs.Connections.Test(r.Context(), cid)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(st); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (s Svc) refreshConnection(w http.ResponseWriter, r *http.Request) {
+	id := r.PathValue("id")
+
+	cid, err := sdktypes.StrictParseConnectionID(id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	st, err := s.svcs.Connections.RefreshStatus(r.Context(), cid)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(st); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	http.Redirect(w, r, fmt.Sprintf("/connections/%s", id), http.StatusFound)
+}
+
+func (s Svc) success(w http.ResponseWriter, _ *http.Request) {
+	fmt.Fprint(w, "success - you can now close this tab.")
```

Comment:
In case we can't close the tab, this message is... underwhelming.

It's preferable to return an HTML page like before, not just a text string. And then it would be up to Ronen to define the look & feel of this page.

Reply:
in case of redir from vscode, we do a redir back there.

other than that i don't mind looks rn.